packages/cli/src/net/createFetch.mts

```diff
@@ -76,6 +77,39 @@ function handleFetchError(e: unknown): Promise<Response> {
     message = "The site version already exists";
   } else if (e.errorName === "VersionNotFound") {
     message = "The site version could not be found";
+  } else if (e.errorName === "VersionLimitExceeded") {
+    const { versionLimit } = e.parameters ?? {};
+    const versionLimitPart = versionLimit != null
+      ? ` (Limit: ${versionLimit} versions)`
+      : "";
+    message = `The site contains too many versions${versionLimitPart}`;
+    tip =
+      "Run the `site version delete` command to delete an old version and try again";
+  } else if (e.errorName === "FileCountLimitExceeded") {
+    const { fileCountLimit } = e.parameters ?? {};
+    const fileCountLimitPart = fileCountLimit != null
+      ? ` (Limit: ${fileCountLimit} files)`
+      : "";
+    message = `The .zip file contains too many files${fileCountLimitPart}`;
+    tip =
+      "Reduce the number of files in the production build to below the limit";
+  } else if (e.errorName === "FileSizeLimitExceeded") {
+    const { currentFilePath, currentFileSizeBytes, fileSizeBytesLimit } =
+      e.parameters ?? {};
+    const currentFilePathPart = currentFilePath != null
+      ? ` "${currentFilePath}"`
+      : "";
+    const currentFileSizePart = currentFileSizeBytes != null
+      ? ` (${prettyBytes(parseInt(currentFileSizeBytes), { binary: true })})`
+      : "";
+    const fileSizeLimitPart = fileSizeBytesLimit != null
+      ? ` (Limit: ${
+        prettyBytes(parseInt(fileSizeBytesLimit), { binary: true })
```

Comment:
Using binary units throughout as the default limit is set in MiB not MB so we display a round whole number